packages/lesswrong/components/community/modules/CommunityMembers.tsx

```diff
@@ -0,0 +1,219 @@
+  },
+  location: {
+    ...theme.typography.commentStyle,
+    color: theme.palette.grey[600],
+    fontSize: 12,
+    fontStyle: 'italic',
+    marginTop: 4,
+  },
+  description: {
+    ...theme.typography.commentStyle,
+    color: theme.palette.grey[800],
+    fontSize: 14,
+    lineHeight: '1.8em',
+    whiteSpace: 'pre-wrap',
+    display: '-webkit-box',
+    "-webkit-line-clamp": 3,
+    "-webkit-box-orient": 'vertical',
+    overflow: 'hidden',
+    marginTop: 12,
+  },
+  map: {
+    [theme.breakpoints.down('sm')]: {
+      display: 'none'
+    },
+  },
+}))
+
+
+const CommunityMembers = ({userLocation, distanceUnit='km', locationFilterNode, classes}: {
+  userLocation: {
+    lat: number,
+    lng: number,
+    known: boolean,
+    loading: boolean,
+  },
+  distanceUnit: 'km'|'mi',
+  locationFilterNode: ReactNode,
+  classes: ClassesType,
+}) => {
+  const { SearchResultsMap } = Components
+  
+  const SearchBox = ({currentRefinement, refine}) => {
+    return <div className={classes.keywordSearch}>
+      <OutlinedInput
+        labelWidth={0}
+        startAdornment={<Search className={classes.searchIcon}/>}
+        placeholder="Search people"
+        value={currentRefinement}
+        onChange={e => refine(e.currentTarget.value)}
+        className={classes.keywordSearchInput}
+      />
+    </div>
+  }
+  const CustomSearchBox = connectSearchBox(SearchBox)
+  
+  const StateResults = ({ searchResults }) => {
+    return (!searchResults || !searchResults.nbHits) ? <div className={classes.noResults}>
+      <div className={classes.noResultsText}>No public profiles matching your search</div>
+    </div> : null
+  }
+  const CustomStateResults = connectStateResults(StateResults)
+  
+  const CommunityMember = ({hit}: {
+    hit: AlgoliaUser,
+  }) => {
+    // the distance from the user's location to the person's location
+    let distanceToPerson;
+    if (userLocation.known && hit._geoloc) {
+      distanceToPerson = `${distance(userLocation, hit._geoloc, distanceUnit)} ${distanceUnit}`
+    }
+    
+    return <div className={classes.person}>
+      <div className={classes.content}>
+        <div className={classes.nameRow}>
+          <Link to={`/users/${hit.slug}`} className={classes.displayName}>{hit.displayName}</Link>
+          <div className={classes.distance}>
+            {distanceToPerson}
+          </div>
+        </div>
+        <div className={classes.location}>{hit.mapLocationAddress}</div>
+        {hit.bio && <div className={classes.description}>{hit.bio}</div>}
+      </div>
+    </div>
+  }
+  
+  // if the user hasn't selected a location, we show the whole map
+  const mapOptions = userLocation.known ? {center: userLocation, zoom: 9} : {zoom: 1}
+  // if the user hasn't selected a location, we just show all users who have a map location (ordered by karma desc)
+  const searchOptions = userLocation.known ? {aroundLatLng: `${userLocation?.lat}, ${userLocation.lng}`} : {filters: "_geoloc.lat>-100"}
```

Comment:
Unfortunately there is no way to filter on `field:exists`, so this is my workaround

Reply:
Love it